src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -438,7 +438,8 @@ R"(#include <stdio.h>
             std::vector<std::string> var_order = ASRUtils::determine_variable_declaration_order(x.m_symtab);
             for (auto &item : var_order) {
                 ASR::symbol_t* var_sym = x.m_symtab->get_symbol(item);
-                if (ASR::is_a<ASR::Variable_t>(*var_sym)) {
+                // var_sym can be nullptr and present in the parent scope.
+                if (var_sym != nullptr && ASR::is_a<ASR::Variable_t>(*var_sym)) {
```

Comment:
Something is wrong here. The `determine_variable_declaration_order` accepts the local symbol table and it is supposed to return only variables from it. So `x.m_symtab->get_symbol(item)` should always succeed. Why doesn't it succeed it here? I think this seems like a bug in `determine_variable_declaration_order`.

Reply:
I'm trying to resolve that in #1378